packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -0,0 +1,93 @@
+import { Aliases, AuthInfo } from '@salesforce/core';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { ForceConfigGet } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+
+export async function setupWorkspaceOrgType() {
+  const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+  const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+  let isScratchOrg = false;
+  if (defaultUsernameIsSet) {
+    const username = await getUsername(defaultUsernameOrAlias!);
+    try {
+      isScratchOrg = await isAScratchOrg(username);
+    } catch (e) {
+      if (e.name === 'NamedOrgNotFound') {
+        // TODO: what should we do in this case? Expose all source commands by default?
+        setDefaultOrgIsScratchOrg(true);
+        setDefaultOrgIsNonScratchOrg(true);
+        return;
+      } else {
+        throw e;
+      }
+    }
+  }
+  setDefaultOrgIsScratchOrg(defaultUsernameIsSet && isScratchOrg);
+  setDefaultOrgIsNonScratchOrg(defaultUsernameIsSet && !isScratchOrg);
+}
+
+async function isAScratchOrg(username: string): Promise<boolean> {
+  const authInfo = await AuthInfo.create(username);
+  const authInfoFields = authInfo.getFields();
+  return Promise.resolve(typeof authInfoFields.devHubUsername !== 'undefined');
+}
+
+/**
+ * Returns the non-aliased username
+ * @param usernameOrAlias
+ */
+async function getUsername(usernameOrAlias: string): Promise<string> {
```

Comment:
How come this one gets a special jsdoc comment?

Reply:
I wasn't sure if it was obvious why I needed to get the username from a username. Would it be better if I removed this, or added jsdoc to all the methods for consistency?